### PR TITLE
refactor: sold and hide refactor

### PR DESCRIPTION
**Pull request by AlefrankM**

### CODE REVIEW

File: src/items/entities/item.entity.ts [hunk 1]
```diff
@@ -49,6 +49,9 @@ export class Item {
   @Column({ nullable: true })
   price: string;
 
+  @Column({ nullable: true })
```
Comment (mmiranda-stackit) [inline src/items/entities/item.entity.ts]: Make all items hidden false by default. Make this column not nullable

File: src/items/enums/item-status.enum.ts [hunk 2]
```diff
@@ -1,8 +1,5 @@
 export enum ItemStatus {
   NotListed = 0,
   Listed = 1,
-  Sold = 2,
   Draft = 3,
```
Comment (mmiranda-stackit) [inline src/items/enums/item-status.enum.ts]: Make Draft = 2

File: src/items/repositories/item.repository.ts [hunk 3]
```diff
@@ -401,31 +401,22 @@ export class ItemRepository extends Repository<Item> {
...
     return item;
   }
 
   private getPaginationConditions(paginationDto: ItemPaginationDto): FindOptionsWhere<Item>[] {
     const { priceFrom, priceTo, search } = paginationDto;
     const conditions = [] as FindOptionsWhere<Item>[];
     const condition1 = {
-      status: Not(In([ItemStatus.Hidden, ItemStatus.Draft])),
+      status: Not(ItemStatus.Draft),
+      isHidden: true,
```
Comment (mmiranda-stackit) [inline src/items/repositories/item.repository.ts]: This should be false

File: src/items/repositories/item.repository.ts [hunk 4]
```diff
@@ -459,7 +450,7 @@ export class ItemRepository extends Repository<Item> {
     const conditions = [] as FindOptionsWhere<Item>[];
     const condition1 = {
       collection: new Collection(collectionId),
-      status: Not(In([ItemStatus.Hidden, ItemStatus.Draft])),
```
Comment (mmiranda-stackit) [inline src/items/repositories/item.repository.ts:462]: You should also avoid pulling drafted items. Please add that condition

**Review by mmiranda-stackit — CHANGES_REQUESTED** [review]

**Review by juliswer — APPROVED** [review]

**Review by mmiranda-stackit — APPROVED** [review]

**Pull request merged**